internal/services/network/application_gateway_resource.go

```diff
@@ -4699,6 +5328,27 @@ func applicationGatewayCustomizeDiff(ctx context.Context, d *pluginsdk.ResourceD
 		}
 	}
 
+	probes := d.Get("probe").(*schema.Set).List()
+	for _, probe := range probes {
+		if probe == nil {
+			continue
+		}
+
+		v := probe.(map[string]interface{})
+		if value, ok := v["protocol"]; ok {
+			protocol := value.(string)
+			if protocol == string(network.ApplicationGatewayProtocolHTTP) || protocol == string(network.ApplicationGatewayProtocolHTTPS) {
+				if pathValue, exists := v["path"]; !exists || len(pathValue.(string)) == 0 {
```

Comment:
This can be:

```suggestion
				if v["path"] == "" {
```

As the key is always existing within the nested block.